src/MongoDB/ClientEncryption.c

```diff
@@ -941,14 +942,15 @@ static void phongo_clientencryption_decrypt(php_phongo_clientencryption_t* clien
 	bson_value_t value      = { 0 };
 	bson_error_t error      = { 0 };
 
-	php_phongo_zval_to_bson_value(zciphertext, PHONGO_BSON_NONE, &ciphertext);
+	phongo_zval_to_bson_value(zciphertext, &ciphertext);
 
 	if (!mongoc_client_encryption_decrypt(clientencryption->client_encryption, &ciphertext, &value, &error)) {
 		phongo_throw_exception_from_bson_error_t(&error);
 		goto cleanup;
 	}
 
-	if (!php_phongo_bson_value_to_zval(&value, zvalue)) {
+	/** Use the legacy decoder to return PHP types instead of BSON types */
```

Comment:
Correct me if I'm mistaken, but this preserves BC and will only differ from the non-legacy function in that Document and PackedArray instances will _not_ be returned. The subtle implication there is that there's no way to obtain a Value instance.

If so, we might want to considering adding an alternative `decrypt()` function that returns a Value directly. Feel free to create a PHPC issue for that if you agree.

Reply:
:point_up: Outstanding question

Reply:
Yes, if we were to use `phongo_bson_value_to_zval` here, any encrypted document or array would suddenly no longer be returned as PHP value but rather as a `PackedArray` or `Document` instance.  To avoid the BC break, I'm using the "legacy" function here.

I created PHPC-2240 to handle this in preparation for the future where we no longer return arrays and `stdClass` instances in PHPC.